src/VisualStudio/Core/Def/DocumentOutline/Readmd.md

```diff
@@ -0,0 +1,28 @@
+# Document Outline
+
+
+## Design
+
+## Implementation Notes
+
+### Queues
+
+We have two queues that we use in document outline: `_documentSymbolQueue` and `_updateViewModelStateQueue`.
+
+Each time the text document is updated the following flow happens:
+
+```mermaid
```

Comment:
:O thank you for introducing this to me, had no idea you could do this and get it properly rendered